File: src/js/_set_doc_attribs.js
////////////////////////////////////////////////////////////////////////////////
// FILE: _set_doc_attribs.js
// AUTHOR: David Ruvolo
// CREATED: 2020-10-07
// MODIFIED: 2020-10-07
// PURPOSE: set document attributes
// DEPENDENCIES: NA
// STATUS: working
// COMMENTS: NA
////////////////////////////////////////////////////////////////////////////////

// define function
function setDocMeta() {
    const targetElem = document.getElementsByTagName("html")[0];
    const refElem = document.getElementById("icecomponents__meta");
    targetElem.lang = refElem.getAttribute("data-html-lang");
    targetElem.dir = refElem.getAttribute("data-html-dir");
}

export default setDocMeta

File: R/set_doc_attribs.R
#' Set Document Attributes
#'
#' Easily set document language and direction
#'
#' @param lang the language the document is written in (ie., ISO639 code)
#' @param dir the direction of the content (e.g., left to right)
#'
#' @examples
#' if (interactive()) {
#'   ui <- tagList(
#'     iceComponents::use_iceComponents(),
#'     iceComponents::set_doc_attribs(lang = "en", dir = "ltr"),
#'     iceComponents::main(
#'       tags$h1("This is my cool app")
#'     )
#'   )
#' }
#' @export
set_doc_attribs <- function(lang = "en", dir = "ltr") {

    stopifnot(
        "`lang` must be a character" = is.character(lang),
        "`dir` is must be 'ltr', 'rtl', or 'auto'" = {
            dir %in% c("auto", "ltr", "rtl")
        }
    )

    tags$span(
        id = "icecomponents__meta",
        style = "display: none;",
        `data-html-lang` = lang,
        `data-html-dir` = dir
    )
}